h2o-py/h2o/frame.py

```diff
@@ -1932,7 +1932,7 @@ def structure(self):
             else:
                 print("num {}".format(" ".join(it[0] if it else "nan" for it in h2o.as_list(self[:10, i], False)[1:])))
 
-    def as_data_frame(self, use_pandas=True, header=True, multi_thread=False):
+    def as_data_frame(self, use_pandas=True, header=True, multi_thread=False, module='datatable'):
```

Comment:
I find the api very confusing now, why not simply have something like:

```python
def as_data_frame(self, header=True, result_format='pandas'):
    """
    :param string result: the format of the data frame, can be one of:
         - list: the data frame is represented as a list of list (native python).
         - pandas: returns a pandas data frame
         - datatable: returns a datable frame (not supported for Py 3.10+)
         - polars: ...
         - arrows: ...
         Note that if speed is a concern (large data), it is recommended to use one of `datatable`, `polars` or `arrows` if they it is already installed in your environment, the resulting dataframe can then easily be converted to `pandas` by calling `to_pandas()` on it.
```

Reply:
note that the `@deprecated_params` annotation would allow to convert the `use_pandas` param to either `result_format='pandas'` or `result_format='list'` and the `multi_thread` param to one of the preferred other format

Reply:
it would also greatly simplify implementation

Reply:
The as_data_frame is just used to convert h2oFrame to pandas dataframe.  The issue here is the user wants it to be faster using multi-thread.  The final result is always a pandas dataframe.  I am just using either datatable or polars/pyarrow to use the multi-thread mode.  If the user just call as_data_frame without anything special, they will get the original single thread mode of conversion.  Only the people in the know will know about multi-thread and the different python toolboxes.

Reply:
> The final result is always a pandas dataframe.

Well, not really, we often use it to convert the h2o frame to python lists as well.

> Only the people in the know will know about multi-thread and the different python toolboxes.

Not if it's mentioned in the documentation that it's preferable to use `polars` for top performance (and then convert to pandas if needed).
I mean with your current changes, you already ask them not only to set the parameter `multi_thread=True` when users just want something faster, they don't care if it will use one or 10 threads. And then on top of that, you ask them to decide on the `module` that will be used as an intermediate to generate this dataframe: it's way too complicated.

My goal is to simplify this, and at the same time provide a more powerful tool that would also convert directly to `polars` or `pyarrow` if the user needs to.

Reply:
Seb:

The work I am working to fix is to convert h2o Frame to Pandas frame only for a customer.  The only thing the user needs to set is multi-thread=True if they want multi-thread.  I want them to use datatable as a default and if they insist, they will choose polar.  Pyarrow does not convert NaN correctly for Python version < 3.9 and hence I am not going to fix it now.

The interface is complicated but necessary to achieve what the customer is looking for.

Reply:
You are right regarding the default output is not always panda frame.  However, the work in the issue is just for transforming h2o frame to panda frame.

Reply:
The use_pandas is a confusing parameter.  The result_format is much better.

Reply:
I deprecated the use_pandas parameter to pandas_frame to denote the output is pandas frame if pandas_frame is true.  I don't know what else to simplify.  Please suggest to me what you are thinking.  Thanks.